src/System.Runtime/tests/System/DateTimeOffsetTests.cs

```diff
@@ -860,14 +860,34 @@ public static void ParseExact_String_String_FormatProvider_DateTimeStyles_G()
             Assert.Equal(expectedString, result.ToString("g"));
         }
 
-        [Fact]
-        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O()
+        [Theory, MemberData("Format_String_TestData_O")]
+        public static void ParseExact_String_String_FormatProvider_DateTimeStyles_O(DateTimeOffset dt, string expected)
         {
-            DateTimeOffset expected = DateTimeOffset.MaxValue;
-            string expectedString = expected.ToString("o");
+            DateTimeOffset result = DateTimeOffset.ParseExact(dt.ToString("o"), "o", null, DateTimeStyles.None);
+            Assert.Equal(expected, result.ToString("o"));
+        }
+
+        public static IEnumerable<object[]> Format_String_TestData_O()
+        {
+            yield return new object[] { DateTimeOffset.MaxValue, "9999-12-31T23:59:59.9999999+00:00" };
+            yield return new object[] { DateTimeOffset.MinValue, "0001-01-01T00:00:00.0000000+00:00" };
+            yield return new object[] { new DateTimeOffset(1906, 8, 15, 7, 24, 5, 300, new TimeSpan(0, 0, 0)), "1906-08-15T07:24:05.3000000+00:00" };
+            yield return new object[] { new DateTimeOffset(1906, 8, 15, 7, 24, 5, 300, new TimeSpan(7, 30, 0)), "1906-08-15T07:24:05.3000000+07:30" };
```

Comment:
I tried to keep this set to a safe set for now. The changes I made in these types were not dependent on any of that and this gets us coverage for the corner cases. I chose the last DateTimeOffset intentionally to have it parse various numbers both single and double digit and made the offset cause a date change.